src/content/docs/build/indexer/txn-stream/txn-filtering.mdx

```diff
@@ -0,0 +1,513 @@
+
+### User Transaction Filter
+
+```rust
+use aptos_transaction_filter::{UserTransactionFilterBuilder, EntryFunctionFilterBuilder, UserTransactionPayloadFilterBuilder};
+
+let filter = UserTransactionFilterBuilder::default()
+    .sender("0x1")
+    .payload(
+        UserTransactionPayloadFilterBuilder::default()
+            .function(
+                EntryFunctionFilterBuilder::default()
+                    .address("0x1")
+                    .module("coin")
+                    .function("transfer")
+                    .build()
+                    .unwrap()
+            )
+            .build()
+            .unwrap()
+    )
+    .build()
+    .unwrap();
+```
+
+### Combining Filters
+
+```rust
+use aptos_transaction_filter::BooleanTransactionFilter;
+
+// Create individual filters
+let success_filter = TransactionRootFilterBuilder::default()
+    .success(true)
+    .build()
+    .unwrap();
+
+let sender_filter = UserTransactionFilterBuilder::default()
+    .sender("0x1")
+    .build()
+    .unwrap();
+
+let event_filter = EventFilterBuilder::default()
+    .struct_type(
+        MoveStructTagFilterBuilder::default()
+            .address("0x1")
+            .module("coin")
+            .build()
+            .unwrap()
+    )
+    .build()
+    .unwrap();
+
+// Combine with logical operators
+let combined = BooleanTransactionFilter::from(success_filter)
+    .or(sender_filter)
+    .and(event_filter);
+
+// Use the filter
+if combined.matches(&transaction) {
+    // Process transaction
+}
+```
+
+### Serialization
+
+```rust
+// Serialize to JSON
+let json = serde_json::to_string_pretty(&filter).unwrap();
+
+// Serialize to YAML
+let yaml = serde_yaml::to_string(&filter).unwrap();
+
+// Deserialize from JSON
+let filter: BooleanTransactionFilter = serde_json::from_str(&json).unwrap();
+```
+
+## Performance Considerations
+
+The transaction filter system is optimized for high-throughput processing:
+
+1. **Single Pass**: Filters process each transaction only once
+2. **Minimal Allocations**: Filters avoid clones and unnecessary copies
+3. **Early Exit**: Filters short-circuit as soon as a non-match is found
+4. **Address Caching**: Address standardization is cached for performance
+
+For best performance:
+- Use specific filters when possible (e.g., filter by address rather than all transactions)
+- Place more restrictive filters first in AND operations
+- Consider the transaction volume on mainnet when designing filters
+
+## Validation
+
+All filters are validated before use. A filter must have at least one field set:
+
+- **TransactionRootFilter**: Must specify `success` or `txn_type`
+- **UserTransactionFilter**: Must specify `sender` or `payload`
+- **EventFilter**: Must specify `struct_type` or `data_substring_filter`
+- **EntryFunctionFilter**: Must specify `address`, `module`, or `function`
+- **MoveStructTagFilter**: Must specify `address`, `module`, or `name`
+
+Invalid filters will return an error when validated or applied.
+
```

Comment:
Especially for the json examples it'd be nice to show them how to actually try it themselves, e.g. with `grpcurl`.

Reply:
<img width="917" height="349" alt="image" src="https://github.com/user-attachments/assets/d02bc6d9-f587-4039-9762-49f1eeca503b" />